### PR TITLE
Added equality operators for Snowflake.

### PR DESCRIPTION
**Pull request by JakenVeina**

### CODE REVIEW

File: Backend/Remora.Discord/Remora.Discord.Core/Snowflake.cs [hunk 1]
```diff
@@ -139,6 +139,24 @@ public int CompareTo(Snowflake other)
...
+            => left.Equals(right);
+
+        /// <summary>
+        /// Compares two snowflakes, for inequality.
+        /// </summary>
+        /// <param name="left">The left operand.</param>
+        /// <param name="right">The right operand.</param>
+        /// <returns>false if the operands are equal, true otherwise.</returns>
+        public static bool operator !=(Snowflake left, Snowflake right)
+            => left.Equals(right);
```
Comment (Tylertron1998) [inline Backend/Remora.Discord/Remora.Discord.Core/Snowflake.cs:158]: I believe you're missing a negation operator here - this does the same thing as `==`

**Review by Tylertron1998 — CHANGES_REQUESTED** [review]

File: Backend/Remora.Discord/Remora.Discord.Core/Snowflake.cs (hunk 1 above)
Comment (JakenVeina) [reply Backend/Remora.Discord/Remora.Discord.Core/Snowflake.cs:158]: Herp.

**Comment by Nihlus** [comment]
Since these operators are new on the type, I'd like you to add some unit tests for them as well. See https://github.com/Nihlus/Remora.Discord/blob/master/Tests/Remora.Discord.Tests/Tests/Core/Snowflake/SnowflakeTests.cs#L193 - you can probably just copy most of that test set as is.

**Comment by JakenVeina** [comment]
I actually was going to add tests, but I didn't see a `Discord.Remora.Core.Tests` project. Didn't realize there were existing tests elsewhere. It might be worth splitting out a separate `Core.Tests` project for clarity.

**Comment by Nihlus** [comment]
Yeah, not a bad idea. As it is, looks good :)

**Pull request merged**